lib/Sabberworm/CSS/Parser.php

```diff
@@ -64,20 +71,91 @@ public function __construct($sText, Settings $oParserSettings = null, $iLineNo =
 			$this->aSizeUnits[$iSize][strtolower($val)] = $val;
 		}
 		ksort($this->aSizeUnits, SORT_NUMERIC);
-	}
+		$this->fixCharset();
+	}
+
+	private function fixCharset() {
+		// We need to know the charset before the parsing starts,
+		// UTF BOMs have the highest precedence and must ve removed before other processing.
+		$this->sOriginalCharset = strtolower($this->oParserSettings->sDefaultCharset);
+		if (strpos($this->sText, self::BOM8) === 0) {
+			$this->sText = substr($this->sText, strlen(self::BOM8));
+			$this->sOriginalCharset = 'utf-8';
+		} else if (strpos($this->sText, self::BOM32BE) === 0) {
+			$this->sText = substr($this->sText, strlen(self::BOM32BE));
+			$this->sOriginalCharset = 'utf-32be';
+		} else if (strpos($this->sText, self::BOM32LE) === 0) {
+			$this->sText = substr($this->sText, strlen(self::BOM32LE));
+			$this->sOriginalCharset = 'utf-32le';
+		} else if (strpos($this->sText, self::BOM16BE) === 0) {
+			$this->sText = substr($this->sText, strlen(self::BOM16BE));
+			$this->sOriginalCharset = 'utf-16be';
+		} else if (strpos($this->sText, self::BOM16LE) === 0) {
+			$this->sText = substr($this->sText, strlen(self::BOM16LE));
+			$this->sOriginalCharset = 'utf-16le';
+		} else if (preg_match('/(.*)@charset\s+["\']([a-z0-9-]+)["\']\s*;/ims', $this->sText, $aMatches)) {
```

Comment:
I disagree here, the presence of BOM is causing problems when processing files in PHP because quite often people forget to ignore it, it is mandatory for utf-16 and utf-32 only